src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml

```diff
@@ -689,31 +689,46 @@
 										</ItemsRepeater.Layout>
 										<ItemsRepeater.ItemTemplate>
 											<DataTemplate x:DataType="filetags:TagViewModel">
-												<StackPanel
-													Height="24"
-													Padding="8,0"
-													HorizontalAlignment="Left"
-													x:Phase="2"
-													BorderBrush="{x:Bind Color, Mode=OneWay, Converter={StaticResource StringToBrushConverter}}"
-													BorderThickness="1"
-													CornerRadius="12"
-													Orientation="Horizontal"
-													Spacing="8"
-													Tapped="TagItem_Tapped"
-													ToolTipService.ToolTip="{x:Bind Name, Mode=OneWay}">
-													<FontIcon
-														FontSize="12"
-														Foreground="{x:Bind Color, Mode=OneWay, Converter={StaticResource StringToBrushConverter}}"
-														Glyph="&#xE1CB;" />
-													<TextBlock
-														x:Name="ItemTag"
-														VerticalAlignment="Center"
-														FontSize="12"
-														LineHeight="18"
-														Style="{StaticResource ColumnContentTextBlock}"
-														Text="{x:Bind Name, Mode=OneWay}"
-														TextTrimming="CharacterEllipsis" />
-												</StackPanel>
+												<UserControl PointerEntered="FileTag_PointerEntered" PointerExited="FileTag_PointerExited">
```

Comment:
Can the PointerEntered event be on the StackPanel?

Reply:
We still would need the `UserControl` to use `VisualStateManager`, this way we save 2 casts

Reply:
Ah right, we can use border though. I think that's preferable here.

Reply:
If you mean using `Border` instead of `UserControl`, unfortunately the problem remains. Border does not inherit from `Control` and hence it can't be used to manage visual state

Reply:
I see, thanks